app/backend/src/couchers/db.py

```diff
@@ -29,10 +34,25 @@ def apply_migrations():
         os.chdir(cwd)
 
 
+@functools.cache
+def get_engine():
+    if config.config["IN_TEST"]:
+        return create_engine(config.config["DATABASE_CONNECTION_STRING"], poolclass=NullPool)
+    else:
+        return create_engine(config.config["DATABASE_CONNECTION_STRING"])
+
+
+@functools.cache
+def get_session_factory():
+    return sessionmaker(bind=get_engine())
+
+
 @contextmanager
-def session_scope(Session):
-    """Provide a transactional scope around a series of operations."""
-    session = Session()
+def session_scope():
+    # get factory...
+    factory = get_session_factory()
+    # ...then call it to get session
+    session = factory()
```

Comment:
get_engine().session(), no point in going a detour through a factory.

Reply:
Does the session factory do no magic?

Reply:
I've pushed a simplification, take a look!